src/components/Editors/FunctionSimulator/Tab.ts

```diff
@@ -0,0 +1,1618 @@
+import { FileTab } from '/@/components/TabSystem/FileTab'
+import { loadAsDataURL } from '/@/utils/loadAsDataUrl'
+import FunctionSimulatorTabComponent from './Tab.vue'
+import { Tab } from '../../TabSystem/CommonTab'
+import { App } from '/@/App'
+import * as monaco from 'monaco-editor'
+import { TabSystem } from '../../TabSystem/TabSystem'
+import Error from './Error.vue'
+import Warning from './Warning.vue'
+import Vue from 'vue'
+import { ca, el, fa, tr } from 'vuetify/src/locale'
+import { BedrockProject } from '../../Projects/Project/BedrockProject'
+import { RefSchema } from '/@/components/JSONSchema/Schema/Ref'
+import { Token } from './Token'
+import { DataLoader } from '../../Data/DataLoader'
+import { isReactive, markRaw } from '@vue/composition-api'
+import { SchemaManager } from '/@/components/JSONSchema/Manager'
+
+export class FunctionSimulatorTab extends Tab {
+	protected fileTab: FileTab | undefined
+	protected currentLine = 0
+	protected content: string = ''
+
+	protected validCommands: Array<string> = []
+	protected validSelectorArgs: Array<string> = []
+	protected commandData: any | undefined
+	protected generalCommandData: any | undefined
+	protected generalSelectorArgsData: any | undefined
+	protected blockStateData: any | undefined
+	protected stopped = false
+
+	constructor(protected parent: TabSystem, protected tab: FileTab) {
+		super(parent)
+		this.fileTab = tab
+	}
+
+	get name(): string {
+		return this.parent.app.locales.translate('functionValidator.tabName')
+	}
+
+	isFor(fileHandle: FileSystemFileHandle): Promise<boolean> {
+		return Promise.resolve(false)
+	}
+
+	component = FunctionSimulatorTabComponent
+
+	get icon() {
+		return 'mdi-cog-box'
+	}
+
+	get iconColor() {
+		return 'primary'
+	}
+
+	save() {}
+
+	protected TranslateError(errorName: string) {
```

Comment:
All method names should start with a lowercase letter (e.g. `translateError` instead of `TranslateError`